### DISCUSSION TITLE
Parameter for static context

### DISCUSSION THREAD

**Discussion by AdrianSosic** [emdgroup/baybe#62] [Ideas]
How about we introduce something like a `FixedParameter` / `ContextParameter` / `StaticParameter` / `ConstantParameter` that allows to explicitly declare data columns with constant values? The motivation for this has been mentioned [here](https://github.com/emdgroup/baybe/pull/6#discussion_r1424248600) and is to reduce the chance of silent bugs. Potentially, there are even more use cases for it.

This new type would gain even more importance if we decide to disallow passing measurement dataframes that carry additional unused columns, which I strongly vote for, because it gives high risk of silent bugs (e.g. if the user simply forgets to declare a parameter).

**Comment by Scienfitz** [comment]
Pro:
- prob easy to implement
- can you elaborate on the additional use cases you see? imo a lot depends on that as for the simple sanity check im not convinced its super useful (see below)

Cons:
- this doesnt come automatic, a user needs to go out their way to define this
- if a user goes out their way, they might as well have a one-liner sanity check before they `add_measurements`
- how will it affect search space creation? If it doesn't affect the searchspace, does it deserve the name/concept of `Parameter` ?
